specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/stable/2025-06-01/diagnostic.json

```diff
@@ -0,0 +1,307 @@
+    "DiagnosticProperties": {
+      "description": "Contains additional properties of a diagnostic",
+      "type": "object",
+      "properties": {
+        "supportedResourceTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Describes what are the supported resource types for a diagnostic."
+        }
+      }
+    },
+    "ErrorResponse": {
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorDetail"
+        }
+      },
+      "description": "An error response from the Compute Diagnostic Resource Provider service."
+    },
+    "ErrorDetail": {
+      "type": "object",
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "The error code."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error."
+        },
+        "message": {
+          "type": "string",
+          "description": "User friendly error message."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorDetail"
+          },
+          "description": "The Api error details"
+        },
+        "innerError": {
+          "$ref": "#/definitions/InnerError",
+          "description": "The Api inner error"
+        }
+      },
+      "description": "Error Detail message."
+    },
+    "InnerError": {
+      "type": "object",
+      "properties": {
+        "exceptiontype": {
+          "type": "string",
+          "description": "The exception type."
+        },
+        "errordetail": {
+          "type": "string",
+          "description": "The internal error message or exception dump."
+        }
+      },
+      "description": "Inner error details."
+    },
+    "SpotPlacementScoresInput": {
+      "type": "object",
+      "properties": {
+        "desiredLocations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The desired regions"
+        },
+        "desiredSizes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceSize"
+          },
+          "description": "The desired resource SKUs."
+        },
+        "desiredCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Desired instance count per region/zone based on the scope."
+        },
+        "availabilityZones": {
+          "type": "boolean",
+          "description": "Defines if the scope is zonal or regional."
+        }
+      },
+      "description": "SpotPlacementScores API Input."
+    },
+    "SpotPlacementScoresResponse": {
+      "type": "object",
+      "properties": {
+        "desiredLocations": {
```

Comment:
>desiredLocations [](http://example.com/codeflow?start=9&length=16)

Do these four properties (desired*, availabilityZones) just reflect the same values that were passed in the request, or are they set by the API?

If they are set by the API, how do they differ from the input values?

If they are just reflecting the input values, I'm not sure it makes sense to have them here in the response.

Reply:
This API has already been rolled out to Prod as part of several preview versions and is being called by several external customers. This PR is just adding a new stable API version to the existing API without any changes to the API request or response entity models